libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFSDKOAuthClient.m

```diff
@@ -586,6 +583,20 @@ - (void)restartAuthentication {
     [self refreshCredentials];
 }
 
+- (SFLoginViewController *) createLoginViewControllerInstance {
+    SFLoginViewController *controller = nil;
+   
+    if (self.config.loginViewControllerConfig.loginViewControllerCreationBlock) {
+        controller = self.config.loginViewControllerConfig.loginViewControllerCreationBlock();
```

Comment:
Do you feel like we may be getting a little too abstract here?  We already have the `authViewDisplayBlock` and dismiss block functionality, for a much more general case.  Should we just make the property a class of `SFLoginViewController` to instantiate?

Reply:
Though it is still typed back to `SFLoginViewController`, so I guess I'm fine with either approach.